codetransformer/transformers/tests/test_mutable_locals.py

```diff
@@ -0,0 +1,152 @@
+"""
+Tests for the mutable_locals transformer.
+"""
+from pytest import raises
+from ..mutable_locals import mutable_locals
+
+
+def test_overwrite():
+
+    @mutable_locals
+    def f():
+        out = []
+        x = 1
+        out.append(x)
+        locals()['x'] = 2
+        out.append(x)
+        return out
+
+    assert f() == [1, 2]
+
+
+def test_create_new():
+
+    @mutable_locals
+    def f():
+        with raises(NameError):
+            y  # noqa
+        locals()['y'] = 1
+        return y  # noqa. Joke's on you pyflakes, this works!
+
+    assert f() == 1
+
+
+def test_assign_after_local_write():
+
+    @mutable_locals
+    def f():
+        out = []
+        locals()['x'] = 1
+        out.append(1)
+        x = 2
+        out.append(x)
+        return out
+
+    assert f() == [1, 2]
+
+
+def test_update():
+
+    @mutable_locals
+    def f():
+        out = []
+        x = 1
+        out.append(x)
+        locals().update({'x': 2, 'y': 3})
+        out.append(x)
+        out.append(y)  # noqa
+        return out
+
+    assert f() == [1, 2, 3]
+
+
+GLOBAL = 1
```

Comment:
small debugging thing, can this be 'GLOBAL' as a str so that when stepping through I know which value I have. then the `2` below becomes 'local' or something?